analyzers/urlanalyzer.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+import re
+import urllib.request
+import urllib.error
+
+from .basicanalyzer import BasicAnalyzer
+
+
+class URLAnalyzer(BasicAnalyzer):
+    """
+    This analyzer detects url-like patterns using regex,
+    and then optionally attempts to verify that they can
+    be resolved.
+    """
+    name = "URLAnalyzer"
+
+    def __init__(self, action, regex, resolve=False):
+        super().__init__(action, regex)
+        self.regex = re.compile(regex)
+        self.resolve = resolve  # Should we try to resolve the URLs?
+
+    def _resolve_url(self, url):
```

Comment:
I love the possibility to try to resolve urls. This is a really useful use-case I have not yet thought about. Well done.